authorizer.py

```diff
@@ -81,19 +81,21 @@ def HasStudentAccess(self):
   def _VerifyServingSession(self, institution, session):
     serving_session = models.ServingSession.FetchEntity(institution)
     logging.info("currently serving session = %s" % serving_session)
-    if serving_session.session_name != session:
-      logging.error("serving session doesn't match")
-      return False
-    if (serving_session.login_type == "schedule" and
-        self.handler.request.path == "/spots_available"):
+    if serving_session.session_name == session:
       return True
-    if (serving_session.login_type == "preregistration" and
-        self.handler.request.path == "/catalog_print"):
+    logging.error("serving session doesn't match")
+    return False
+
+  def HasPageAccess(self, institution, session, current_page):
```

Comment:
Actually I just saw how you are using this. It looks pretty good actually. The only thing I might change is I might remove the current_page parameter. Instead I'd pass back the current serving level. Then I'd let the page decide if it should show itself based on the serving level.